RoslynPluginGenerator/AnalyzerPluginGenerator.cs

```diff
@@ -371,6 +410,29 @@ private void GenerateFixedSqaleFile(IEnumerable<DiagnosticAnalyzer> analyzers, s
             logger.LogDebug(UIResources.APG_SqaleGeneratedToFile, outputFilePath);
         }
 
+        /// <summary>
+        /// Checks that the supplied rule file has valid content
+        /// </summary>
+        private bool IsValidRulesFile(string filePath)
+        {
+            Debug.Assert(!string.IsNullOrWhiteSpace(filePath));
```

Comment:
IMO those 2 assertions needs to be `if/throw`.

Reply:
They're assertions because they are checking the validity of our code, not the user input. We don't expect our code to reach this point if the file is null or missing.
If the user has supplied a valid file then the code will already have thrown (see comment in line 419 below).

Reply:
Ok.